test/jdk/java/awt/ScrollPane/bug8077409Test.java

```diff
@@ -31,87 +31,109 @@
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+import java.awt.Toolkit;
+import java.awt.event.KeyEvent;
 
 public class bug8077409Test extends Frame {
-  ScrollPane pane;
-  MyCanvas myCanvas;
+    ScrollPane pane;
+    MyCanvas myCanvas;
 
-  class MyCanvas extends Canvas {
-    public Dimension getPreferredSize() {
-      return new Dimension(400, 800);
-    }
+    public bug8077409Test() {
+        super();
+        setLayout(new BorderLayout());
+        pane = new ScrollPane();
+
+        myCanvas = new MyCanvas();
+        pane.add(myCanvas);
+
+        add(pane, BorderLayout.CENTER);
+        setSize(320, 480);
 
-    public void paint(Graphics g) {
-      g.setColor(Color.BLACK);
-      g.drawLine(0, 0, 399, 0);
-      g.setColor(Color.RED);
-      g.drawLine(0, 1, 399, 1);
-      g.setColor(Color.BLUE);
-      g.drawLine(0, 2, 399, 2);
-      g.setColor(Color.GREEN);
-      g.drawLine(0, 3, 399, 3);
     }
 
-  }
-
-  public bug8077409Test() {
-    super();
-    setLayout(new BorderLayout());
-    pane = new ScrollPane();
-
-    myCanvas = new MyCanvas();
-    pane.add(myCanvas);
-
-    add(pane, BorderLayout.CENTER);
-    setSize(320, 480);
-
-  }
-
-  @Override
-  protected void processKeyEvent(KeyEvent e) {
-    super.processKeyEvent(e);
-
-  }
-
-  public static void main(String[] args) throws AWTException, InterruptedException {
-    final bug8077409Test obj = new bug8077409Test();
-    obj.setVisible(true);
-    Toolkit.getDefaultToolkit().addAWTEventListener(new AWTEventListener() {
-      @Override
-      public void eventDispatched(AWTEvent e) {
-        KeyEvent keyEvent = (KeyEvent) e;
-        if(keyEvent.getID() == KeyEvent.KEY_RELEASED) {
-            if (keyEvent.getKeyCode() == KeyEvent.VK_1) {
-              System.out.println(obj.pane.toString());
-              System.out.println("obj.myCanvas.pos: " + obj.myCanvas.getBounds());
-              System.out.println(obj.myCanvas.toString());
-            }  else if (keyEvent.getKeyCode() == KeyEvent.VK_2) {
-              obj.repaint();
-           } else if (keyEvent.getKeyCode() == KeyEvent.VK_DOWN) {
-              Point scrollPosition = obj.pane.getScrollPosition();
-              scrollPosition.translate(0, 1);
-              obj.pane.setScrollPosition(scrollPosition);
-            } else if (keyEvent.getKeyCode() == KeyEvent.VK_UP) {
-              Point scrollPosition = obj.pane.getScrollPosition();
-              scrollPosition.translate(0, -1);
-              obj.pane.setScrollPosition(scrollPosition);
-            } else if (keyEvent.getKeyCode() == KeyEvent.VK_SPACE) {
-              obj.pane.validate();
+    public static void main(String[] args) throws AWTException, InterruptedException {
+        final bug8077409Test obj = new bug8077409Test();
+        try {
+            obj.setLocationRelativeTo(null);
+            Toolkit.getDefaultToolkit().addAWTEventListener(e -> {
```

Comment:
This test is not manual and does not use a robot to click. How can this listener be called?

Reply:
Yea, that code piece was unwanted, I have removed it and kept it in a clean way.

I have also tested this with the latest changes on Windows-x64 with JDK 8u60_b04 and JDK 8u60_b20 as the original issue related to this test(JDK-8077409) was raised for Windows-x64 and was fixed in JDK 8u60b20.
Results:

With JDK 8u60b04:
Exception in thread "main" java.lang.RuntimeException: Wrong position of component in ScrollPane
        at bug8077409Test.main(bug8077409Test.java:142)

With JDK 8u60b20:
Passed.....